### PR TITLE
refactor(kommander): Use thanos & karma source code as subcharts

### PR DESCRIPTION
**Pull request by samvantran**
## What changes are proposed in this pull request?

Resolves: [D2IQ-63905](https://jira.d2iq.com/browse/D2IQ-63905)

This PR copies the chart src code from [kommander-thanos](https://github.com/mesosphere/charts/tree/master/stable/kommander-thanos) and [kommander-karma](https://github.com/mesosphere/charts/tree/master/stable/kommander-karma) into kommander's `charts/` folder. This will help reduce the amount of PRs needed when updating the charts.

Previously if we were to make a change in, say, Karma we'd have to:
- create a PR to update the kommander-karma code and bump karma version
- create a PR to bump the [kommander chart](https://github.com/mesosphere/charts/tree/master/stable/kommander) version and update karma tarball
- create a PR in [kubeaddons-kommander](https://github.com/mesosphere/kubeaddons-kommander) repo to bump the kommander chart version

By porting the kommander-karma and -thanos code, we can reduce by one step and make updates _and_ bump the kommander chart version in the same PR. This speeds up development and reduces PR reviews for code changes that were already approved.

**Notes**: 
- A lot of files shown in the PR but it's really just copy+paste. The only changes I made were in the last two commits. 
- I'm mimicking [istio subcharts](https://github.com/mesosphere/charts/tree/master/staging/istio/charts) 
- The code is ready but I'm marking as `wip` for now until post-Kommander GA 

## How were these changes tested?
- same way as described in https://github.com/mesosphere/charts/pull/433 except I deployed Konvoy with cluster.yaml
  ```yaml
    - configRepository: https://github.com/samvantran/kubeaddons-kommander
      configVersion: combine-central-monitoring
  ```
  - checked Thanos and Karma addons deployed normally and dashboards were up and running 
- ran [this Konvoy CI job](https://teamcity.mesosphere.io/viewLog.html?buildId=2600255&tab=buildResultsDiv&buildTypeId=ClosedSource_MkeLite_MkeLiteE2e) to test CI against the addon changes

### CODE REVIEW

**Comment by shaneutt** [comment]
@samvantran please don't hesitate to ask @mesosphere/sig-ksphere-catalog for any assistance you might need moving this from `wip` to `ready`!

**Comment by shaneutt** [comment]
@samvantran just wanted to ping on this one and see if any help is needed from @mesosphere/sig-ksphere-catalog ?

**Comment by juliangieseke** [comment]
CI says no :( 
> Error: requirements.lock is out of sync with requirements.yaml

File: stable/kommander/charts/kommander-karma/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (juliangieseke) [inline stable/kommander/charts/kommander-karma/Chart.yaml]: actually, this is out of date.

File: stable/kommander/charts/kommander-thanos/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+appVersion: "1.0"
```
Comment (juliangieseke) [inline stable/kommander/charts/kommander-thanos/Chart.yaml]: same here.

**Comment by samvantran** [comment]
> Sorry for not catching this earlier: shouldnt the old ones being removed in the same PR? Like actually moved instead of copied?

Didn't want to do too much in one PR. If it's all right with you, would rather make that a separate task.

**Pull request merged**